lib/semantics/tools.h

```diff
@@ -471,5 +471,79 @@ FindPolymorphicAllocatableUltimateComponent(const DerivedTypeSpec &);
 UltimateComponentIterator::const_iterator
 FindPolymorphicAllocatableNonCoarrayUltimateComponent(const DerivedTypeSpec &);
 
+class LabelEnforce {
+public:
+  LabelEnforce(SemanticsContext &context, std::set<parser::Label> &&labels,
+      parser::CharBlock constructSourcePosition, const char *construct)
+    : context_{context}, labels_{labels},
+      constructSourcePosition_{constructSourcePosition}, construct_{construct} {
+  }
+  template<typename T> bool Pre(const T &) { return true; }
+  template<typename T> bool Pre(const parser::Statement<T> &statement) {
+    currentStatementSourcePosition_ = statement.source;
+    return true;
+  }
+
+  template<typename T> void Post(const T &) {}
+
+  void Post(const parser::GotoStmt &gotoStmt) { checkLabelUse(gotoStmt.v); }
+  void Post(const parser::ComputedGotoStmt &computedGotoStmt) {
```

Comment:
Member function definitions should go into .cc files when possible.  Please move the definition to tools.cc

Reply:
Done.